frontend/src/pages/students/[id]/register-communication.tsx

```diff
@@ -15,23 +26,144 @@ export default function CommunicationInfoPage() {
     const router = useRouter();
     const query = router.query as { id: string };
     const studentId = query.id;
+    const { data: receivedTemplates, error: templateError } = useSWR(
+        apiPaths.communicationTemplates,
+        getAllCommunicationTemplatesFromPage
+    );
+    const [selectedTemplate, setSelectedTemplate] = useState<ICommunicationTemplate | undefined>(undefined);
+    const [create, setCreate] = useState(false);
+    const [edit, setEdit] = useState(false);
+    const templates: ICommunicationTemplate[] = receivedTemplates || [];
 
     const { data: receivedStudent, error: studentError } = useSWR(
         apiPaths.students + "/" + studentId,
         getStudentOnUrl
     );
 
-    if (studentError) {
-        console.log(studentError);
+    if (studentError || templateError) {
+        console.log(studentError || templateError);
         return null;
     }
 
     const student: IStudent = receivedStudent || emptyStudent;
 
     return (
         <>
-            <h1>{capitalize(t("register communication for")) + " " + student.callName}</h1>
-            <CreateCommunicationForm student={student} />
+            <NavBar />
```

Comment:
Can you extract this int o a component as well? This page is now realllyyy big :)